framework/Models/EnsembleModel.py

```diff
@@ -467,12 +467,20 @@ def collectOutput(self,finishedJob,output):
         for key in exportDict['metadata']:
           output.updateMetadata(key,exportDict['metadata'][key][-1])
     # collect outputs for "holding"
-    if "holdOutputErase" in exportDict['metadata']:
-      if exportDict['metadata']['holdOutputErase'] is not None:
-        keys = self.tempOutputs['forHold'].keys()
-        for key in keys:
-          if key.startswith(exportDict['metadata']['holdOutputErase'][0]):
-            self.tempOutputs['forHold'].pop(key)
+    # first clear old outputs
+    # TODO FIXME this is a flawed implementation, since it requires that the "holdOutputErase" is of
+    # a very specific form that works with the the current SPSA optimizer.  Since we have no other optimizer right now,
+    # the problem is only extensibility, not the actual implementation.
+    if exportDict['metadata'].get('holdOutputErase',None) is not None:
+      #if exportDict['metadata']['holdOutputErase'] is not None:
+      keys = self.tempOutputs['forHold'].keys()
+      toErase = exportDict['metadata']['holdOutputErase'][0].split('_')[:2]
+      for key in keys:
+        traj,itr,pert = key.split('_')
+        if traj == toErase[0] and itr <= toErase[1]:
+          self.tempOutputs['forHold'].pop(key)
```

Comment:
For the record, I ran a script to test the timing of these two operations:
```python
from timeit import timeit as t

def const():
  return {'b':{'c':6}}

def runDel():
  a = const()
  del a['b']['c']

def runPop():
  a = const()
  a['b'].pop('c')

d = t(runDel,number=1000000)
p = t(runPop,number=1000000)

print 'Seconds for del:',d
print 'Seconds for pop:',p
```
The results on my macbook was:
```
Seconds for del: 0.488843917847
Seconds for pop: 0.568461179733
```